### PR TITLE
fix(style): keep checkbox.group style the same as v4

### LINKED ISSUE #41950: Checkbox.Group 排版样式与 Antd v4 保持一致
### What problem does this feature solve?

Checkbox.Group 候选项太多，外层容器固定宽度的情况下。

问题一
v5 版本的 group 使用 `display: inline-flex`，候选项会挤在一行，不会换行，文字竖着排版
v4 版本的 group 使用 `display: inline-block`，会自动换行

问题二
v5 版本候选项采用左边距间隔，换行后第二行会导致出现多于的空白区域

### What does the proposed API look like?

Checkbox.Group 排版样式与 Antd v4 保持一致

### PR DESCRIPTION
**Pull request by BoyYangzai**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- fix https://github.com/ant-design/ant-design/issues/41950

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    fix(style): keep checkbox.group style the same as v4       |
| 🇨🇳 Chinese |     checkbox.group style 风格与v4保持一致      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 1afca5d</samp>

Improved the styling of `checkbox` and `checkbox.Group` components to support flex and horizontal layouts. Modified `components/checkbox/style/index.ts` to adjust the display and margin properties.

### <samp>🤖 Generated by Copilot at 1afca5d</samp>

* Fix layout issue of checkbox inside flex container by changing `display` to `inline-block` and adding margin-right to checkbox group items ([link](https://github.com/ant-design/ant-design/pull/42103/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL38-R45))
* Remove margin-inline-start from checkbox wrapper to avoid unnecessary margins when checkbox group is aligned horizontally ([link](https://github.com/ant-design/ant-design/pull/42103/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL57-L61))

### CODE REVIEW

File: components/checkbox/style/index.ts [hunk 1]
```diff
@@ -35,7 +35,14 @@ export const genCheckboxStyle: GenerateStyle<CheckboxToken> = (token) => {
...
         ...resetComponent(token),
 
-        display: 'inline-flex',
+        display: 'inline-block',
+
+        // Group-Item
+        [`${checkboxCls}-group-item`]: {
+          ...resetComponent(token),
+
+          marginRight: '8px',
```
Comment (MadCcc) [inline components/checkbox/style/index.ts:44]:
```suggestion
          marginInlineEnd: token.marginXS
```

**Review by li-jia-nan — APPROVED** [review]

File: components/checkbox/style/index.ts [hunk 2]
```diff
@@ -36,6 +36,13 @@ export const genCheckboxStyle: GenerateStyle<CheckboxToken> = (token) => {
         ...resetComponent(token),
 
         display: 'inline-flex',
+        flexWrap: 'wrap',
+        columnGap: '8px',
```
Comment (MadCcc) [inline components/checkbox/style/index.ts:40]: 用 token 哈

Comment (MadCcc) [inline components/checkbox/style/index.ts:40]: 用 token 哈

**Comment by MadCcc** [comment]
image
这分的太开了吧？

File: components/checkbox/style/index.ts [hunk 3]
```diff
@@ -36,6 +36,13 @@ export const genCheckboxStyle: GenerateStyle<CheckboxToken> = (token) => {
...
+
+        // Group > Grid
+        '.ant-row': {
```
Comment (MadCcc) [inline components/checkbox/style/index.ts:43]: v4 没有这个样式吧

File: components/checkbox/style/index.ts [hunk 4]
```diff
@@ -36,6 +36,13 @@ export const genCheckboxStyle: GenerateStyle<CheckboxToken> = (token) => {
...
+          width: '100%',
+        },
```
Comment (MadCcc) [inline components/checkbox/style/index.ts:45]:
```suggestion
        [`${token.antCls}-row`]: {
          flex: 1,
        },
```
1. `ant` 前缀要用 token，这是动态的
2. flex 布局下用 `flex: 1` 撑满

Comment (BoyYangzai) [reply components/checkbox/style/index.ts:45]: ok

**Comment by MadCcc** [comment]
image
这个 diff 是没有用 token 导致 compact 主题下间距不对，改了就好了。
这个 PR 应该只影响 grid 布局的 snapshot

**Comment by BoyYangzai** [comment]
> image 这个 diff 是没有用 token 导致 compact 主题下间距不对，改了就好了。 这个 PR 应该只影响 grid 布局的 snapshot

get

**Review by MadCcc — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Pull request merged**